### ISSUE TITLE
Investigate architecture for Blockchain Integration

### ISSUE THREAD

**Issue by maevyn11** [GridPlus/phonon-client#77]
In order to support deposit, redeem, and validate functionality in the phonon-client we need to investigate which APIs most cleanly support our use cases. 

We also need to determine where messaging for each of these functionalities takes place, whether it should happen on the client or in the frontend.

**Comment by maevyn11** [comment]
I found a couple of block explorer APIs that provide multichain access including BTC and ETH but it seems their their API's do not include any unified abstraction layer that would let you query comparable information from both with one data structure. Basically, we would have to write distinct modules for these assets anyway and they would just request from the same backend, which might save on costs but doesn't really save us any code complexity.

The two I found are TokenView and Blockchairs, both with time limited free tiers.
This sort of makes sense to me because the primitives in UTXO based chains and account based chains are represented pretty differently. A general purpose unified API that could hit both would probably be a leaky abstraction.

Basically this exercise has helped me realize that we are going to need to build a specific phonon abstraction that supports individual asset deposit, validation, and redemption out from phonons on the basis of blockchain models and then individual chains.

So in effect the API backend is really just a modular component that we could get from one or many providers as we see fit. We can just build it with the most expedient option right now and swap it out for something more complex later if we have problems or find opportunities for simplification.
